dspace/config/dspace.cfg

```diff
@@ -72,19 +72,42 @@ db.password = dspace
 #      so this may be set to ${db.username} in most scenarios.
 db.schema = public
 
-## Connection pool parameters
+## Database Connection pool parameters
+# DSpace wraps around the Apache Commons DBCP2 component, to read about its configuration
+# see: https://commons.apache.org/proper/commons-dbcp/configuration.html
 
-# Maximum number of DB connections in pool (default = 30)
+# Maximum number of active DB connections in pool (-1 = unlimited)
+# (default = 30)
 db.maxconnections = 30
 
-# Maximum time to wait before giving up if all connections in pool are busy (milliseconds)
+# Maximum time to wait before giving up if all connections in pool are busy (milliseconds), (-1 = unlimited)
 # (default = 5000ms or 5 seconds)
 db.maxwait = 5000
 
+# Minimum number of idle connections in pool
+# (default = 0)
+db.minidle = 0
+
 # Maximum number of idle connections in pool (-1 = unlimited)
 # (default = 10)
 db.maxidle = 10
 
+# The initial number of connections created when the pool is started
+# (default = 0)
+db.initialsize = 0
+
+# The maximum lifetime in milliseconds of a connection.  (-1 = unlimited)
+# (default = -1)
+db.maxconnlifetime = -1
+
+# Remove abandoned connections.
```

Comment:
We should document that this affects only removeAbandonedOnBorrow, not removeAbandonedOnMaintenance.

Reply:
Yes, good spot.

My reasoning was mostly to keep things simple. Using removeAbandonedOnMaintenance requires yet another property, timeBetweenEvictionRunsMillis, to be set. This then initiates an "evictor thread" to be run at intervals to evict idle connection. I'll be frank and say I have no idea how this is supposed to work, let alone the interplay between the removeAbandoned process and the evictor process.

Where should I document this? In the config file or the wiki or both?

Reply:
Yes, removeAbandonedOnMaintenance is probably expert territory.  I don't think we need to cater for it in normal DSpace configuration.  Someone who wants to adjust it can do so directly in `config/spring/api/core-hibernate.xml`.

We should simply note in the config file comment that this affects `removeAbandonedOnBorrow`.  We can explain more fully in the wiki.